app/assets/src/components/AlignmentViz.jsx

```diff
@@ -0,0 +1,276 @@
+import axios from "axios";
+import React from "react";
+
+class ReadViz extends React.Component {
+  constructor(props) {
+    super(props);
+    this.parsedAlignment = this.parseAlignment(props);
+  }
+
+  parseAlignment(readInfo) {
+    let readPart = parseInt(readInfo.name.split("/")[1])
+    let reversed = 0
+    let seq_len = readInfo.sequence.length
+    let sequence = readInfo.sequence
+
+
+    readInfo.metrics[0] = parseFloat(readInfo.metrics[0])
+    for (let i = 1; i <= 7; i+=1) {
+      readInfo.metrics[i] = parseInt(readInfo.metrics[i])
+    }
+    readInfo.metrics[8] = parseFloat(readInfo.metrics[8])
+    readInfo.metrics[9] = parseFloat(readInfo.metrics[9])
+
+    if (readInfo.metrics[6] > readInfo.metrics[7]) {
+      reversed = 1
+      sequence = sequence.split("").reverse().join("");
+    }
+
+    if ((reversed == 1 && readPart == 1) || (reversed == 0 && readPart == 2)) {
+      let m4 = readInfo.metrics[4]
+      let m5 = readInfo.metrics[5]
+      readInfo.metrics[4] = seq_len - m5 + 1
+      readInfo.metrics[5] = seq_len - m4 + 1
+    }
+
+    let aligned_portion = sequence.slice(readInfo.metrics[4] - 1, readInfo.metrics[5])
+    let left_portion = (readInfo.metrics[4] - 2) >= 0 ? sequence.slice(0, readInfo.metrics[4] - 1) : ""
+    let right_portion = readInfo.metrics[5] < seq_len ? sequence.slice(readInfo.metrics[5]) : ""
+
+    if (readInfo.refInfo[0].length > left_portion.length) {
+      // trim refInfo[0]
+      readInfo.refInfo[0] = readInfo.refInfo[0].slice(readInfo.refInfo[0].length - left_portion.length)
+    } else {
+      // pad refInfo[0]
+      while(readInfo.refInfo[0].length < left_portion.length) {
+        readInfo.refInfo[0] = ' ' + readInfo.refInfo[0]
+      }
+    }
+
+    if (readInfo.refInfo[2].length > right_portion.length) {
+      //trim refInfo[2]
+      readInfo.refInfo[2] = readInfo.refInfo[2].slice(0, right_portion.length)
+    } else {
+      while(readInfo.refInfo[2].length < right_portion.length) {
+        readInfo.refInfo[2] += ' '
+      }
+    }
+    let [quality_string1, mis_matches1] = this.generateQualityString(readInfo.refInfo[1], aligned_portion)
+    let [quality_string2, mis_matches2] = this.generateQualityString(readInfo.refInfo[1], this.complementSeq(aligned_portion))
+    let quality_string = mis_matches1 < mis_matches2 ? quality_string1 : quality_string2
+    let white_space_left = this.repeatStr(' ', left_portion.length)
+    let white_space_right = this.repeatStr(' ', right_portion.length)
+    let ref_seq_display = readInfo.refInfo.join("|")
+    let read_seq_display = [left_portion, aligned_portion, right_portion].join("|")
+    let quality_string_display = [white_space_left, quality_string, white_space_right].join("|")
+
+    // generate quality string
+    return {
+      reversed: reversed,
+      read_part: readPart,
+      ref_display: ref_seq_display,
```

Comment:
snake_case to camelCase :D!

Reply:
Yes! Airbnb actually has a really decent JavaScript style guide that influences a lot of companies' own style guides, such as Google.

https://github.com/airbnb/javascript/blob/master/README.md#objects

The object section shows that the keys use lowerCamelCase in general. But then again, each project can have their own preference!